docker/compose.local.yml

```diff
@@ -1,34 +1,92 @@
+    ports:
+      - "80:80"
     volumes:
-      - "../deploy/config/client/env.local.js:/dtaas/client/build/env.js"
+      - /var/run/docker.sock:/var/run/docker.sock
+
   libms:
-    image: intocps/libms
+    image: intocps/libms:latest
+    volumes:
+      - ${DTAAS_DIR}/deploy/config/lib.docker:/dtaas/libms/.env
+      - ${DTAAS_DIR}/files:/dtaas/libms/files
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.libms.entryPoints=web"
+      - "traefik.http.services.libms.loadbalancer.server.port=4001"
+      - "traefik.http.routers.libms.rule= PathPrefix(`/lib`)"
+      - "traefik.http.routers.libms.middlewares=traefik-forward-auth"
+
+  client:
+    image: intocps/dtaas-web:latest
     volumes:
-      - "../deploy/config/lib.docker:/dtaas/libms/.env"
-      - "../files:/dtaas/libms/files"
+      - ${ENV_JS_FILEPATH}:/dtaas/client/build/env.js
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.client.entryPoints=web"
+      - "traefik.http.services.client.loadbalancer.server.port=4000"
+      - "traefik.http.routers.client.middlewares=traefik-forward-auth"
+      - "traefik.http.routers.client.rule=PathPrefix(`/`)"
 
-  ml-workspace-user1:
+  user1:
     image: mltooling/ml-workspace-minimal:0.13.2
-    container_name: ml-workspace-user1
     volumes:
-      - "../files/user1:/workspace"
-      - "../files/common:/workspace/common"
+      - ${DTAAS_DIR}/files/common:/workspace/common
+      - ${DTAAS_DIR}/files/${username1}:/workspace
     environment:
-      - AUTHENTICATE_VIA_JUPYTER
-      - WORKSPACE_BASE_URL=user1
-    shm_size: 512m
+      - AUTHENTICATE_VIA_JUPYTER=
+      - WORKSPACE_BASE_URL=${username1}
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.u1.entryPoints=web"
+      - "traefik.http.routers.u1.rule=PathPrefix(`/${username1}`) "
+      - "traefik.http.routers.u1.middlewares=traefik-forward-auth"
+  
+  user2:
+    image: mltooling/ml-workspace-minimal:0.13.2
+    volumes:
+      - ${DTAAS_DIR}/files/common:/workspace/common 
+      - ${DTAAS_DIR}/files/${username2}:/workspace
+    environment:
+      - AUTHENTICATE_VIA_JUPYTER=
+      - WORKSPACE_BASE_URL=${username2}
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.u2.entryPoints=web"
+      - "traefik.http.routers.u2.rule=PathPrefix(`/${username2}`) "
+      - "traefik.http.routers.u2.middlewares=traefik-forward-auth"
 
-  traefik-gateway:
-    image: traefik:v2.10
-    container_name: traefik-gateway
-    ports:
-      - "9000:80"
+  traefik-forward-auth:
+    image: thomseddon/traefik-forward-auth:latest
     volumes:
-      - "../deploy/config/gateway/traefik.yml:/etc/traefik/traefik.yml"
-      - "../deploy/config/gateway/auth:/etc/traefik/auth"
-      - "../deploy/config/gateway/dynamic/fileConfig.docker.yml:/etc/traefik/dynamic/fileConfig.yml"
-      - "/var/run/docker.sock:/var/run/docker.sock"
\ No newline at end of file
+      - ${OAUTH_CONF_FILEPATH}:/conf
```

Comment:
given that `conf.allow` is only useful for compose.local.yml, we might as well hardcode it here.